azext_iot/monitor/parsers/central_parser.py

```diff
@@ -92,17 +92,29 @@ def _perform_dynamic_validations(self, payload: dict):
         if not isinstance(template, Template):
             return
 
-        # pnp device is sending data to an unrecognized interface
-        if self.interface_name and (self.interface_name not in template.interfaces):
-            details = strings.invalid_interface_name(
-                self.interface_name, list(template.interfaces.keys())
+        # if component name is not defined then data should be mapped to root/inherited interfaces
+        if not self.component_name:
+            self._validate_payload(
+                payload=payload, template=template, is_component=False
             )
+            return
+
+        if not template.components:
+            # tempate does not have any valid components
+            details = strings.invalid_component_name(self.component_name, list())
```

Comment:
We should have a separate error here - one where it tells the user the template doesn't have any components

Reply:
error mapping is per PM recommendation. Error message will have an empty list indicating that there are no components defined in template.